### PR TITLE
FIX: lit-path!  or get-path! with ":" be treated as set-path!

### PR DESCRIPTION
**Pull request by bitbegin**
For example: 'a/b: 1 for :a/b: 1 should not be compiled ok

### CODE REVIEW

**Comment by bitbegin** [comment]
before: 
```
a: context [b: 0]
probe 'a/b
probe 'a/b: 1

probe :a/b
probe :a/b: 2

```

will be allowed.

After merge this PR:

```
Red []

a: context [b: 0]
probe 'a/b
probe a/b: 1

probe :a/b
probe a/b: 2

```

only this way be compiled ok.

**Comment by greggirwin** [comment]
We certainly want tests to go along with this, as it's an important, syntactic decision. A `lit-set-path!` has no effect, and a `get-set-path!` is already handled by `set-path!` (just talking about the interpreter and design, not compiler issues). That is, if the final selector results in an active value, like a func, it's not evaluated if a `set-path!` is used.

**Comment by dockimbel** [comment]
Those syntactic forms should not be `load`-able (even though they could be created programmatically), as they have confusing meaning.

The fact that they were allowed so far should be considered a bug and this PR is a bugfix, restricting the lexer to only valid syntaxes.

@greggirwin If you have no objections, we can accept this PR.

**Comment by greggirwin** [comment]
No objections here.

**Comment by greggirwin** [comment]
Though appveyor issue doesn't look like a timeout.

**Comment by bitbegin** [comment]
it passed on Rebol 2.7.8 , failed on 2.7.6 (but this version not response on my PC). 

Seems run-all-comp1.exe crashed, when used "-d" flag, this can be reproduced (because of `--assert error? try [1 + if false [2]]`). (but strange with this PR)

**Comment by greggirwin** [comment]
I'll let you and @dockimbel make the call on merging it then, just in case.

File: environment/lexer.red [hunk 1]
```diff
@@ -624,9 +624,10 @@ system/lexer: context [
 					  reject
 				]
 			]
-			opt [#":" (type: set-path! set-path back tail stack)][
-				ahead [path-end | ws-no-count | end] | (throw-error [type path])
+			opt [#":" (type: set-path! set-path back tail stack)
+				(if wtype <> word! [throw-error [type path]])
```
Comment (bitbegin) [inline environment/lexer.red]: if change to: 
```
(type: set-path! set-path back tail stack if wtype <> word! [throw-error [type path]])
```
test case will be failed on Rebol 2.7.6

**Comment by greggirwin** [comment]
Ping. OK to merge?

**Comment by bitbegin** [comment]
@greggirwin can't merge, as i still can't solve the ci issue

**Comment by greggirwin** [comment]
Guess we'll wait for @qtxie or @dockimbel to assist. Thanks.

**Comment by dockimbel** [comment]
Closing this PR, as it has been replaced by another fix (referenced above).

**Pull request closed without merging**